src/charm.py

```diff
@@ -59,6 +131,11 @@ def _application_config(self) -> dict | None:
                 return None
             application_config = json.loads(config)
             if isinstance(application_config, dict):
+                if "spring" not in application_config:
```

Comment:
`application_config` could be converted to a `defaultdict` with the `dict` constructor to make this easier

Reply:
Changed